lab_12/src/lib.c

```diff
@@ -0,0 +1,58 @@
+#include <lib.h>
+#include <stdio.h>
+
+char *strconcat(char *str1, char *str2)
+{
+    char *result = str1;
+
+    // While no endline
+    while (*str1 != '\0')
+    {
+        str1++;
+    }
+
+    // While no endline copy str2 to str1
+    while (*str2 != '\0')
+    {
+        *str1 = *str2;
+        str1++;
+        str2++;
+    }
+
+    // Add endline to terminate the string
+    *str1 = '\0';
+    return result;
+}
```

Comment:
1) Empty strings?
2) Not enough space in `str1`?
---
I believe it is a buggy function.